qiskit/pulse/instructions/play.py

```diff
@@ -31,31 +33,98 @@ class Play(Instruction):
     cycle time, dt, of the backend.
     """
 
-    def __init__(self, pulse: Pulse, channel: PulseChannel, name: Optional[str] = None):
+    def __init__(
+        self,
+        pulse: Pulse,
+        mixed_frame: Optional[Union[PulseChannel, MixedFrame]] = None,
+        pulse_target: Optional[PulseTarget] = None,
+        frame: Optional[Frame] = None,
```

Comment:
It seems like there is an abstraction here missing for the core IR, ie., there are too many possible inputs? We should not focus on the flexibility of input in the object structure but rather have this in the builder interface/compiler passes themselves.

For example, are mixed_frame/ports all PulseTarget?

Reply:
This was something I went back and forth with about `PulseTarget` - should `MixedFrames` (and `Channel`) be `PulseTarget`? It has some merit, for example, here it will simplify the arguments. But later on, you'll always have to check if your `PulseTarget` is a `MixedFrame`\ `Channel` or a `Port`\ `LogicalElement` - which behave differently. For this reason, I decided to keep `PulseTarget` only for things which will eventually map to a port (and don't mix the frame in).

The option of `mixed_frame` is added here as a syntactic sugar, and it might be better to just remove it.